s3torchbenchmarking/src/s3torchbenchmarking/benchmark.py

```diff
@@ -71,6 +75,27 @@ def from_conf(dataset_config: DictConfig):
             return DatasetSharding[dataset_config.sharding]
 
 
+def make_mountpoint(
+    prefix_uri: str,
+    mountpoint_path: Optional[str] = None,
+    additional_args: Optional[str] = None,
+) -> str:
+    def teardown(path: str):
+        subprocess.run(["sudo", "umount", path])
+        shutil.rmtree(path)
+
+    bucket, prefix = parse_s3_uri(prefix_uri)
+    # Run Mountpoint in background mode, and arrange for it to unmount when this script exits
+    tempdir = tempfile.mkdtemp(prefix="s3dataset_")
+    binary = mountpoint_path or "mount-s3"
+    args = additional_args.split(" ") if additional_args else []
+    subprocess.run([binary, bucket, tempdir] + args, check=True)
+    atexit.register(teardown, tempdir)
+
+    # Now we can just read our dataset as if it were a local directory
+    return str(Path(tempdir) / prefix)
+
+
 def make_dataset(
```

Comment:
Non-blocking: We need to simply this factory method.

Reply:
Sure. What simplification did you have in mind?

Reply:
non-blocking: At the first glance, extract separately the logic for each kind, so that here we have just a method call - i.e.
```
if kind == "s3iterabledataset":
    create_iterable_dataset(...) 
```
As `make_dataset` will grow with the addition of supported dataset types, but the logic for each individual type can be separated.